src/boardState.ts

```diff
@@ -0,0 +1,239 @@
+                    
+                    if((!shotImpliedPlacementsList[j].resolved)&& Position.isEqual(shotImpliedPlacementsList[j].shot.Position, occupiedPositions[i])){
+                        
+                        numberUnresolved--;
+                        shotImpliedPlacementsList[j].resolved = true;
+                    }
+                    
+                }
+                shotImpliedPlacementsList[j].updateRemainingShips(numberRemaining);
+                shotImpliedPlacementsList[j].removeInvalidPositions(validShipPositions);
+            }
+
+            shotImpliedPlacementsList.sort((a,b,)=>{return a.allValidPlacements.length - b.allValidPlacements.length});
+        }
+
+
+        //for whatever ships have yet to be placed (needs to be checked)
+        //place those ships randomly
+
+        
+
+        for(var i=0; i<shipLengths.length; i++){
+            if(numberRemaining[shipLengths[i]]>0){
+                let nextShip: ShipPlacement;
+
+
+                if(validShipPositions[i].numberOfPossibilities>0){
+                    if(!exhaustive){
+                        nextShip = validShipPositions[i].pickRandomPlacement();
+                    }
+                    else{
+                        let nextStep: number = 0;
+                        if(pathToFollow.length > currentPathIndex){
+                            nextStep = pathToFollow[currentPathIndex];
+                            if(nextStep > validShipPositions[i].numberOfPossibilities-1){
+                                let nextPath:number[] = currentPath;
+                                nextPath[nextPath.length-1]++;
+                                return {state: new BoardState([]),found:false, nextPath: nextPath};
+                            }
+                        }
+                        currentPathIndex++;
+                        currentPath.push(nextStep);
+                        nextShip = validShipPositions[i].pickSpecificPlacement(nextStep);
+                    }
+                }
+                else{
+                    if(!exhaustive)
+                    {
+                        return {state: new BoardState([]), found:false, nextPath: []};
+                    }else{
+                        let nextPath:number[] = currentPath;
+                        nextPath[nextPath.length-1]++;
+                        return {state: new BoardState([]),found:false, nextPath: nextPath};
+                    }
+                }
+                
+                for(var j=0; j<validShipPositions.length; j++){
+                    validShipPositions[j].removePossibilities(nextShip);
+                }
+                possibleShips.push(nextShip);
+                numberRemaining[shipLengths[i]]--;
+            }
+        }
+        let nextPath:number[] = currentPath;
+        nextPath[nextPath.length-1]++;
+        return {state: new BoardState(possibleShips), found:true, nextPath}
```

Comment:
If you're returning a bunch of this stuff tied together, why not make a specific class for it?